compiler/rustc_mir/src/transform/promote_consts.rs

```diff
@@ -478,7 +478,7 @@ impl<'tcx> Validator<'_, 'tcx> {
                                         // https://github.com/rust-lang/rust/pull/74945#discussion_r463063247
                                         // There may be opportunity for generalization, but this needs to be
                                         // accounted for.
-                                        if proj_base.is_empty()
+                                        if place_base.projection.is_empty()
```

Comment:
I think the `is_empty` check should be moved up near `self.temps[place_base.local]` -- that code only makes sense if there are no projections.